soc/arm/atmel_sam/common/soc_poweroff.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 Bjarki Arge Andreasen
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/poweroff.h>
+#include <soc.h>
+
+#ifndef SUPC_CR_KEY_PASSWD
+#define SUPC_CR_KEY_PASSWD (0xA5 << 24)
+#endif
```

Comment:
usase of ifndef here seems to hide some header issues, can you please explain?

Reply:
The headers for the atmel sam chips in `modules/hal/atmel/asf` are irregular, defining the SUPC_CR_KEY_PASSWD only for some chips, this guard ensures the value is not redefined, and is included in this form for the remaining chips.

Examples of defines from different headers for the same hardware:
```
#define SUPC_CR_KEY(value) ((SUPC_CR_KEY_Msk & ((value) << SUPC_CR_KEY_Pos)))
```
```
#define   SUPC_CR_KEY_PASSWD (0xA5u << 24)
```

Reply:
does this value depend on SoC?

Reply:
No, its constant :) Just wanted to reuse the name for clarity, I could just not use it

Reply:
If HAL is inconsistent I'd just define it with a similar name locally and forall, so no need for ifndef hackery.

Reply:
Hacked off ifndef hackery